scripts/data/gen-nodetime/src/relayer/lib/relayer.ts

```diff
@@ -0,0 +1,433 @@
+	rpcAddr: string;
+	addrPrefix: string;
+	gasPrice: string;
+};
+type ConnectOptions = {
+	sourcePort: string;
+	sourceVersion: string;
+	targetPort: string;
+	targetVersion: string;
+	ordering: "ORDER_UNORDERED" | "ORDER_ORDERED";
+};
+type Connections = {
+	srcConnection: string;
+	destConnection: string;
+};
+type Endpoint = {
+	channelID?: string;
+	chainID: string;
+	portID: string;
+};
+type Path = {
+	id: string;
+	isLinked: boolean;
+	src: Endpoint;
+	dst: Endpoint;
+};
+type LinkResponse = {
+	linkedPaths: string[];
+	alreadyLinkedPaths: string[];
+};
+type StartResponse = {};
+type PacketHeights = {
+	packetHeightA: number;
+	packetHeightB: number;
+	ackHeightA: number;
+	ackHeightB: number;
+};
+type PathConfig = {
+	path: Path;
+	options?: ConnectOptions;
+	connections?: Connections;
+	relayerData?: PacketHeights;
+};
+type RelayerConfig = {
+	mnemonic: string;
+	chains?: Array<ChainConfig>;
+	paths?: Array<PathConfig>;
+};
+
+export default class Relayer {
+	public config: RelayerConfig;
+	public homedir: string;
+	public configFile: string;
+	private pollTime: number;
+	private relayers: Map<string, ReturnType<typeof setInterval>>;
+
+	constructor(configFile: string = "config.yaml", pollTime = 5000) {
+		this.homedir = os.homedir();
+		this.pollTime = pollTime;
+		this.configFile = configFile;
+		this.relayers = new Map();
+		this.config = this.readOrCreateConfig();
+		const nestedProxy = {
+			set: (target, prop, value) => {
+				target[prop] = value;
+				this.writeConfig(this.config);
+				return true;
+			},
+			get: (target, prop) => {
+				if (typeof target[prop] === "object" && target[prop] !== null) {
+					return new Proxy(target[prop], nestedProxy);
+				} else {
+					return target[prop];
+				}
+			},
+		};
+		this.config = new Proxy(this.config, nestedProxy);
+	}
+	createConfigFolder() {
+		try {
+			if (!fs.existsSync(this.getConfigFolder())) {
+				fs.mkdirSync(this.getConfigFolder());
+			}
+		} catch (e) {
+			throw new Error(Errors.configFolderFailed + e);
+		}
+	}
+	getConfigFolder() {
+		return this.homedir + "/.ts-relayer";
+	}
+	getConfigPath() {
+		return this.getConfigFolder() + "/" + this.configFile;
+	}
+	readOrCreateConfig() {
+		this.createConfigFolder();
+		try {
+			if (fs.existsSync(this.getConfigPath())) {
+				let configFile = fs.readFileSync(this.getConfigPath(), "utf8");
+				return yaml.load(configFile);
+			}
+			let config = {
+				mnemonic: Bip39.encode(Random.getBytes(32)).toString(),
+			};
+			let configFile = yaml.dump(config);
+			fs.writeFileSync(this.getConfigPath(), configFile, "utf8");
+			return config;
+		} catch (e) {
+			throw new Error(Errors.configReadFailed + e);
+		}
+	}
+	writeConfig(config) {
+		try {
+			let configFile = yaml.dump(config);
+			fs.writeFileSync(this.getConfigPath(), configFile, "utf8");
+		} catch (e) {
+			throw new Error(Errors.configWriteFailed + e);
+		}
+	}
+
+	// RPC Handlers
+	public async ensureChainSetup([rpcAddr, gasPrice, addrPrefix]: [
```

Comment:
when the chain exists, can we make options (gasPrice, addrPrefix) updated in the config.yml?

Reply:
can we also support the scnerio where I configure two chains with `relayer conf` and then decide to change the chainid in one of these and re-run `relayer conf` with success (right now it complains about another chain already being exist with the same rpc-chainid pair).

Reply:
1st one done...
2nd one I don't think should be a feature as we only get the chain id from the endpoint (it is not supplied by the user) and an endpoint suddenly serving a different chain id is not something we should hide from the user